build-parent/pom.xml

```diff
@@ -0,0 +1,289 @@
+              <id>check-assert-imports</id>
+              <phase>process-sources</phase>
+              <goals>
+                <goal>enforce</goal>
+              </goals>
+              <configuration>
+                <rules>
+                  <RestrictImports>
+                    <includeTestCode>true</includeTestCode>
+                    <reason>Use AssertJ for Assertions instead of JUnit</reason>
+                    <bannedImports>
+                      <bannedImport>static org.junit.jupiter.api.Assertions.*</bannedImport>
+                      <bannedImport>org.junit.jupiter.api.Assertions</bannedImport>
+                      <bannedImport>static org.junit.Assert.*</bannedImport>
+                      <bannedImport>org.junit.Assert</bannedImport>
+                    </bannedImports>
+                  </RestrictImports>
+                  <RestrictImports>
+                    <includeTestCode>true</includeTestCode>
+                    <reason>Use AssertJ for Assertions instead of Hamcrest</reason>
+                    <bannedImports>
+                      <bannedImport>static org.hamcrest.MatcherAssert.*</bannedImport>
+                      <bannedImport>org.hamcrest.MatcherAssert</bannedImport>
+                    </bannedImports>
+                  </RestrictImports>
+                </rules>
+                <fail>true</fail>
+              </configuration>
+            </execution>
+          </executions>
+        </plugin>
+
+        <plugin>
+          <groupId>net.revelc.code.formatter</groupId>
+          <artifactId>formatter-maven-plugin</artifactId>
+          <version>${formatter.plugin.version}</version>
+          <dependencies>
+            <dependency>
+              <artifactId>quarkus-ide-config</artifactId>
+              <groupId>io.quarkus</groupId>
+              <version>${quarkus.formatter.version}</version>
+            </dependency>
+          </dependencies>
+          <configuration>
+            <!-- store outside of target to speed up formatting when mvn clean is used -->
+            <cachedir>.cache/formatter-maven-plugin-${formatter-maven-plugin.version}</cachedir>
+            <configFile>eclipse-format.xml</configFile>
+            <lineEnding>LF</lineEnding>
+            <skip>${formatter.skip}</skip>
+          </configuration>
+          <executions>
+            <execution>
+              <goals>
+                <goal>${formatter.goal}</goal>
+              </goals>
+            </execution>
+          </executions>
+        </plugin>
+        <plugin>
+          <groupId>net.revelc.code</groupId>
+          <artifactId>impsort-maven-plugin</artifactId>
+          <version>${impsort.plugin.version}</version>
+          <configuration>
+            <!-- store outside of target to speed up formatting when mvn clean is used -->
+            <cachedir>.cache/impsort-maven-plugin-${impsort-maven-plugin.version}</cachedir>
+            <groups>java.,javax.,jakarta.,org.,com.</groups>
```

Comment:
What about `io.`?
```suggestion
            <groups>java.,javax.,jakarta.,org.,com.,io.</groups>
```

Reply:
I'm using the same configuration of the quarkus-ide-config plugin https://github.com/quarkusio/quarkus/blob/dbd7370d21193108195469e4fe55d9455ede8aef/build-parent/pom.xml#L690

Reply:
The developer's IDE has to be configured with those files, I would not change the configuration as we would have to refer to that module for the codestyle and to another config file for the imports in this repo for example.